### PR TITLE
Robotic skin research page

### PR DESCRIPTION
**Pull request by alecive**
Creating the PR so that I can see the deployed version and add comments at some point

### CODE REVIEW

**Comment by alecive** [comment]
Right now is kinda early, but lemme know if and when I should give feedback.

**Comment by calebescobedo** [comment]
The first version of the robotic skin research page is ready. Please take a look at it!

**Comment by watakandai** [comment]
any update?

**Comment by calebescobedo** [comment]
I added two potential future work projects. Please take a look at them and give me some feedback if you would like me to change anything @watakandai

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+title: Robotic Skin
+description: Research Page
+tags: [robotics,panda,simulator,research]
+permalink: robotic_skin.html
+---
+
+{% include image.html url="research/skin_unit.png" max-width="100%" description="Flexible skin unit created by etching the circuit diagram into a copper sheet and then encasing it in the flexible polymer ecoflex." %}
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: This "skin unit" thing you keep talking about is not a standard concept, nor it doesn't make much sense. If you have a whole-body skin, what does it mean to have a skin unit? I don't understand why you refer to this as skin unit as if it was as mainstream as the word pixel or gradient descent. I would rephrase parts of this document and using those two words as common words rathrr than specific official words (and this is true also for the paper)

Comment (alecive) [reply _posts/wiki/2020-03-23-robotic-skin.md]: For what concerns this caption: you should also add a description of the capability, which is missing as of now.

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+Robots have been transitioning into human-populated environments and replacing their physical cages used to keep humans safe with complex perception and software. However, current solutions are computationally expensive, prone to occlusion, and require setup overhead. Robots require compact, self-contained sensing of nearby space to provide inherently save interactions. In this work, we present the first prototype of a flexible, conformant whole-body artificial skin for collaborative robotics. In order to locate all skin units on the surface of a robot, a novel kinematically calibration algorithm is required to reduce setup time.
+
+# Contributions
+
+1. <strong>Kinematic Calibration:</strong> We present a framework for automatic kinematic calibration that leverages an Inertial Measurement Unit (IMU) to automatically locate skin units, of arbitrary number, along the surface of a robot.
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: this is not how you do bold in markdown

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+title: Robotic Skin
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: better title needed

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+description: Research Page
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: better subtitle needed

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+tags: [robotics,panda,simulator,research]
+permalink: robotic_skin.html
+---
+
+{% include image.html url="research/skin_unit.png" max-width="100%" description="Flexible skin unit created by etching the circuit diagram into a copper sheet and then encasing it in the flexible polymer ecoflex." %}
+
+
+Robots have been transitioning into human-populated environments and replacing their physical cages used to keep humans safe with complex perception and software. However, current solutions are computationally expensive, prone to occlusion, and require setup overhead. Robots require compact, self-contained sensing of nearby space to provide inherently save interactions. In this work, we present the first prototype of a flexible, conformant whole-body artificial skin for collaborative robotics. In order to locate all skin units on the surface of a robot, a novel kinematically calibration algorithm is required to reduce setup time.
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: > "In order to locate all skin units on the surface of a robot, a novel kinematically calibration algorithm is required to reduce setup time."

This comes out of nowhere and needs explanation. Nobody can understand how you jump from the previous sentence to this.

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+# Contributions
+
+1. <strong>Kinematic Calibration:</strong> We present a framework for automatic kinematic calibration that leverages an Inertial Measurement Unit (IMU) to automatically locate skin units, of arbitrary number, along the surface of a robot.
+
+1. <strong>Skin Unit:</strong> We present a new <strong>flexible artificial skin</strong> that can conform to multiple surfaces. The skin utilizes an IMU and a proximity sensor to sense obstacles at a distance. Future prototypes will embed additional capabilities (e.g. pressure sensing, temperature sensing, tactile sensing...)
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: the skin unit contribution should be presented before the kinematic calibration (so that you can also introduce what do you mean with SU)

Comment (alecive) [reply _posts/wiki/2020-03-23-robotic-skin.md]: Same with the sections below

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+ 
+{% include image.html url="research/calibration.png" max-width="75%" description="Illustrated setup of skin units (S) placed along the robots links (L) that are separated by joints (J). We estimate the  Denavit-Hartenberg parameters of each joint in order to calculate the pose of each skin unit along the surface of the robot. " %}
+
+# Skin Units
+Skin units are embedded with an IMU for kinematic calibration and a proximity sensor to sense nearby objects. We chose to embed our circuit in a flexible polymer called ecoflex in order to have each skin unit conform to the shape of the surface it is placed on. The figure below shows the flexible capabilities of the skin sensor and an image of one unit mounted on a robotic arm.
+
+{% include image.html url="research/skin_unit.jpg" max-width="50%" description="Flexible skin unit mounted on a robotic arm." %}
+
+{% include image.html url="research/bent_skin_unit.jpg" max-width="50%" description="Demonstration of the skin’s flexible capabilities." %}
+
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: Please look at this file https://raw.githubusercontent.com/alecive/alecive.github.io/master/_posts/research/humanoid_robotics/2015-08-01-yarpdroid.md rendered here https://alessandro.ronc.one/research/2015/08/01/yarpdroid/ to get pictures side by side

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+With this information, we apply a repulsive force perpendicular to the skin unit as seen in the figure below.
+
+{% include image.html url="research/control.png" max-width="75%" description=" A mounted skin unit exerting a repulsive force β on a robot's trajectory 
+<strong>T</strong> when an object is near the skin unit. The distance d is measured in millimeters. As an object approaches, β increases exponentially." %}
+
+
+# Future Work
+
+### 1. Dynamic Obstical Avoidance
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: "obstical" lol

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+# Kinematic Calibration
+To automatically locate skin units along the surface of a robot, we utilize an IMU that provides measurements of angular velocity and linear acceleration. Our optimization algorithm estimates the position and orientation of skin units using Denavit-Hartenberg as demonstrated in the figure below.
+ 
+{% include image.html url="research/calibration.png" max-width="75%" description="Illustrated setup of skin units (S) placed along the robots links (L) that are separated by joints (J). We estimate the  Denavit-Hartenberg parameters of each joint in order to calculate the pose of each skin unit along the surface of the robot. " %}
+
+# Skin Units
+Skin units are embedded with an IMU for kinematic calibration and a proximity sensor to sense nearby objects. We chose to embed our circuit in a flexible polymer called ecoflex in order to have each skin unit conform to the shape of the surface it is placed on. The figure below shows the flexible capabilities of the skin sensor and an image of one unit mounted on a robotic arm.
+
+{% include image.html url="research/skin_unit.jpg" max-width="50%" description="Flexible skin unit mounted on a robotic arm." %}
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: These are the same pictures as before. Does it make sense to have them again?

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+# Contributions
+
+1. <strong>Kinematic Calibration:</strong> We present a framework for automatic kinematic calibration that leverages an Inertial Measurement Unit (IMU) to automatically locate skin units, of arbitrary number, along the surface of a robot.
+
+1. <strong>Skin Unit:</strong> We present a new <strong>flexible artificial skin</strong> that can conform to multiple surfaces. The skin utilizes an IMU and a proximity sensor to sense obstacles at a distance. Future prototypes will embed additional capabilities (e.g. pressure sensing, temperature sensing, tactile sensing...)
+
+1. <strong>Control Example:</strong> We demonstrate an example control scenario where a robot arm utilizes artificial skin to <strong>avoid collision</strong>  with a human, increasing operational safety with and around people.
+
+
+# Kinematic Calibration
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: It doesn't explain what it does. It'd be nice to have some equations

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+{% include image.html url="research/calibration.png" max-width="75%" description="Illustrated setup of skin units (S) placed along the robots links (L) that are separated by joints (J). We estimate the  Denavit-Hartenberg parameters of each joint in order to calculate the pose of each skin unit along the surface of the robot. " %}
+
+# Skin Units
+Skin units are embedded with an IMU for kinematic calibration and a proximity sensor to sense nearby objects. We chose to embed our circuit in a flexible polymer called ecoflex in order to have each skin unit conform to the shape of the surface it is placed on. The figure below shows the flexible capabilities of the skin sensor and an image of one unit mounted on a robotic arm.
+
+{% include image.html url="research/skin_unit.jpg" max-width="50%" description="Flexible skin unit mounted on a robotic arm." %}
+
+{% include image.html url="research/bent_skin_unit.jpg" max-width="50%" description="Demonstration of the skin’s flexible capabilities." %}
+
+# Control Example
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: Same here: it is too lame and it doesn't explain anything useful. I need techincal details.

File: _posts/wiki/2020-03-23-robotic-skin.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+* Calibrated pose of the Skin Unit (x, y, z, φ, θ, ψ)
+* Distance d normal to the skin sensor
+
+With this information, we apply a repulsive force perpendicular to the skin unit as seen in the figure below.
+
+{% include image.html url="research/control.png" max-width="75%" description=" A mounted skin unit exerting a repulsive force β on a robot's trajectory 
+<strong>T</strong> when an object is near the skin unit. The distance d is measured in millimeters. As an object approaches, β increases exponentially." %}
+
+
+# Future Work
```
Comment (alecive) [inline _posts/wiki/2020-03-23-robotic-skin.md]: 1. I would flip the two things below (hardware goes always first)
2. You are just explaining the two papers, could be worth using this as an opportunity to start thinking about how to _extend_ them
3. Could be also worth having some ideas for future projects for incoming students

**Comment by alecive** [comment]
Ok I have added my review. Some of this changes are minor, some are bigger. Most importantly, some of them should not be handled by @calebescobedo alone, but all the people on the @HIRO-group/roboskin  project should be involved and contributing to it!!

File: _posts/wiki/2020-03-23-robotic-skin.md (hunk 3 above)
Comment (calebescobedo) [reply _posts/wiki/2020-03-23-robotic-skin.md:2]: Maybe the poster title: **Self-Contained Kinematic Calibration of Flexible Whole-Body Artificial Skin for Collaborative Robotics**? @watakandai @peasant98 any other ideas for a compelling title?

File: _posts/wiki/2020-03-23-robotic-skin.md (hunk 4 above)
Comment (calebescobedo) [reply _posts/wiki/2020-03-23-robotic-skin.md:3]: How about **Outline of the three main artificial skin contributions: skin unit design, automatic kinematic calibration, and collision avoidance**

File: _posts/wiki/2020-03-23-robotic-skin.md (hunk 10 above)
Comment (calebescobedo) [reply _posts/wiki/2020-03-23-robotic-skin.md]: @watakandai Are the error equations that we used in the poster session still accurate to use in this part of the research page? If not, what else do you think we should use in this section to express how calibration is done? 

Screenshot from 2020-04-05 15-17-02

Comment (watakandai) [reply _posts/wiki/2020-03-23-robotic-skin.md]: I don't

**Comment by alecive** [comment]
Nice improvements. Getting there. I still need details though!

**Comment by alecive** [comment]
I think one thing that is missing right now is the broader vision. Right now the article talks about your current work, but it'd be nice to have a broader vision for this in the beginning. Also the title is too specific: it should be a more general, all-encompassing title for all the papers that come out of the roboskin project.

File: _posts/research/2020-04-05-robotic-skin.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+    {% include image.html url="research/roboskin/bent_skin_unit.jpg" max-width="75%" description="Demonstration of the skin’s flexible capabilities." %}
+  </div>
+</div>
+
+
+## Kinematic Calibration
+
+To automatically locate skin units along the surface of a robot, we utilize an IMU that provides measurements of angular velocity and linear acceleration. Our optimization algorithm estimates the position and orientation of skin units using Denavit-Hartenberg as demonstrated in the figure below.
+
+{% include image.html url="research/roboskin/calibration.png" max-width="75%" description="Illustrated setup of skin units (S) placed along the robots links (L) that are separated by joints (J). We estimate the  Denavit-Hartenberg parameters of each joint in order to calculate the pose of each skin unit along the surface of the robot. " %}
```
Comment (watakandai) [inline _posts/research/2020-04-05-robotic-skin.md:56]: @alecive  Do we need to describe the method more in detail?

File: _posts/research/2020-04-05-robotic-skin.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+
+
+## Control Example
+
+Calibrated skin units can be used to locate obstacles with respect to the robot’s links and avoid undesired collisions by exerting a repulsive force on the robotic arm. To avoid object we require the following information:
+
+* Calibrated pose of the Skin Unit (x, y, z, φ, θ, ψ)
```
Comment (watakandai) [inline _posts/research/2020-04-05-robotic-skin.md:64]: position (x, y, z) and quaternion (x, y, z, w)

Comment (watakandai) [reply _posts/research/2020-04-05-robotic-skin.md:64]: @alecive  Are we able to use MathJax?

File: _posts/research/2020-04-05-robotic-skin.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+* Distance d normal to the skin sensor
+
+With this information, we apply a repulsive force perpendicular to the skin unit as seen in the figure below.
+
+{% include image.html url="research/roboskin/control.png" max-width="75%" description=" A mounted skin unit exerting a repulsive force β on a robot's trajectory
+**T** when an object is near the skin unit. The distance d is measured in millimeters. As an object approaches, β increases exponentially." %}
```
Comment (watakandai) [inline _posts/research/2020-04-05-robotic-skin.md:70]: Why is $\beta$ applied to 2 directions ? 
In addition, the coordinate systems should be separated. The $world$ frame should be better attached to the robot's base and the IMU coordinate should be placed close to the IMU. So that way we know which frame refers to which object. It would be also good to have a description in here.

File: _posts/research/2020-04-05-robotic-skin.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+
+# Future Work
+
+
+## Dense Coverage of a Robot Arm with Skin Units
```
Comment (watakandai) [inline _posts/research/2020-04-05-robotic-skin.md:76]: We should divide this into 2 sections. One for addressing the dense coverage problem and the other for addressing the flexibility and the stretch-ability of the skin.

## A Short-Term Future Work
### Dense Coverage of a Robot Arm with Skin Units
...
### Flexibility and Stretchability of a robotic skin
...

### Dynamic Obstacle Avoidance 
....


## Long-Term Future Work
Here, address our vision. 

Better to have an image for each section.

File: _posts/wiki/2020-03-23-robotic-skin.md (hunk 10 above)
Comment (watakandai) [reply _posts/wiki/2020-03-23-robotic-skin.md]: I strongly disagree to post equations and our current method online even though we haven't published our paper.
Recently, I heard from one of my friend that his method was stolen from his co-author (professor). We should not address any of our method here on the website.

File: _posts/research/2020-04-05-robotic-skin.md (hunk 13 above)
Comment (alecive) [reply _posts/research/2020-04-05-robotic-skin.md:56]: Yes, I think that another paragraph on how this works below the picture would be great!

File: _posts/research/2020-04-05-robotic-skin.md (hunk 14 above)
Comment (alecive) [reply _posts/research/2020-04-05-robotic-skin.md:64]: I remember I tried in the past but it wasn't working well, I can definitely check it again at some point (it'd be nice to have anyways). Maybe we can keep mathjax formulas in a future iteration of this page.

File: _posts/research/2020-04-05-robotic-skin.md (hunk 15 above)
Comment (alecive) [reply _posts/research/2020-04-05-robotic-skin.md]: agreed

File: _posts/research/2020-04-05-robotic-skin.md (hunk 16 above)
Comment (alecive) [reply _posts/research/2020-04-05-robotic-skin.md:86]: I'd like to have something on the vision in the very beginning--it should be the first thing the reader reads. If they get it, they get everything else! 

Also, feel free to take inspiration from the article that came out in december @calebescobedo !

File: _posts/wiki/2020-03-23-robotic-skin.md (hunk 10 above)
Comment (watakandai) [reply _posts/wiki/2020-03-23-robotic-skin.md]: @alecive What do you think?

Comment (alecive) [reply _posts/wiki/2020-03-23-robotic-skin.md]: Ok then, let's wait to add equations then. I would still like to have some technical content though--any thoughts on how to make this happen?

**Comment by alecive** [comment]
@calebescobedo is it ready for me to review? I'd also like to remove the interactive

**Comment by calebescobedo** [comment]
@alecive Yes, with Kandai's changes it should be ready to review!

**Comment by alecive** [comment]
Ok guys the article is in great shape and very close to be released. @watakandai @calebescobedo @aran159 please take a look at my commit above and look at the diff because it could be giving you some hints on how to improve writing stylistically and be overall more concise and effective. I'm far from being the best, but it could be useful experience regardless!! 

I left three comments directly in the file for you guys to take care of---they are very minor things. Once you fix them, let me know and we can merge!

File: _posts/research/2020-04-22-robotic-skin.md [hunk 17]
```diff
@@ -8,163 +8,128 @@ image:
...
+    $${}^0T_{SU_i} = {}^0T_1 \cdot {}^1T_2 \dots {}^{i-1}T_i \cdot {}^i T_{SU_i}, \quad \forall i$$, where
+    $$
+    {}^{i}T_{i+1} =
+        \left[ \begin{array}{c|c}
+            {}^{i}R_{i+1} & {}^{i}P_{i+1} \\
+            \hline
+            \mathbf{0} & \mathbf{1} \\
+        \end{array}\right].
+    $$
+    **[AR: I don't think the rotation part of this is correct---I know we've been through this already but it needs revision. I just removed it for the sake of compactness]**{:.color-banner}
```
Comment (watakandai) [inline _posts/research/2020-04-22-robotic-skin.md:73]: Could you give me a reason why this notation is wrong?

**Comment by watakandai** [comment]
Thank you @alecive. I noticed many changes were made. Your sentence makes things much simpler and easier to read. I need to improve my English more. :+1:

File: _posts/research/2020-04-22-robotic-skin.md (hunk 17 above)
Comment (alecive) [reply _posts/research/2020-04-22-robotic-skin.md:73]: Yes! But maybe we should take this offline.

Comment (calebescobedo) [reply _posts/research/2020-04-22-robotic-skin.md:73]: @watakandai did y'all end up resolving this? Can you remove the part of the equation that we don't use?

**Comment by alecive** [comment]
Ok guys, I have reviewed the doc and added minor changes. To me, it's good to go! Good job, I really like it! 
Given that I can't Merge myself because I have opened the PR, I give you the honor :-)
